utils/id_hash.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"bytes"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	u "github.com/ipfs/go-ipfs-util"
+	blake2b "golang.org/x/crypto/blake2b"
+)
+
+// ID signifies that its contents have been hashed from either a
+// peer.ID or a util.Key. This unifies the keyspace.
+type ID []byte
```

Comment:
The difference is insignificant

byte.Compare function takes a slice. To convert that to array, I have used copy(not sure if this could be done any other way), which could make it slower. 

benchmarks here
https://gist.github.com/kishansagathiya/d47935941e649a563e8f18a54ba6ec9b

Reply:
I don't see where you need to convert slices to arrays. You need to convert arrays to slices which does not require copy.